test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/testproviders/OptaplannerTestProvider.java

```diff
@@ -111,24 +112,24 @@ public void testExecuteSolver(KieServerDeployment kieServerDeployment,
         KieServicesClient kieServerClient = KieServerClientProvider.getKieServerClient(kieServerDeployment, extraClasses(kieContainer));
 
         SolverServicesClient solverClient = kieServerClient.getServicesClient(SolverServicesClient.class);
-        SolverInstance solverInstance = solverClient.createSolver(containerId,
-                                                                  CLOUD_BALANCE_SOLVER_ID, CLOUD_BALANCE_SOLVER_CONFIG);
-        Assertions.assertThat(solverInstance).isNotNull();
-
-        Object planningProblem = loadPlanningProblem(kieContainer, 5, 15);
-        solverClient.solvePlanningProblem(containerId, CLOUD_BALANCE_SOLVER_ID, planningProblem);
-
-        solverInstance = solverClient.getSolver(containerId, CLOUD_BALANCE_SOLVER_ID); // wait for 15 seconds
-        for (int i = 0; i < 5 && solverInstance.getStatus() == SolverInstance.SolverStatus.SOLVING; i++) {
-            Thread.sleep(3000);
-            solverInstance = solverClient.getSolver(containerId, CLOUD_BALANCE_SOLVER_ID);
-        }
-
-        solverInstance = solverClient.getSolver(containerId, CLOUD_BALANCE_SOLVER_ID);
-        Assertions.assertThat(solverInstance.getStatus()).isEqualTo(SolverInstance.SolverStatus.NOT_SOLVING);
-        Assertions.assertThat(solverInstance.getScoreWrapper()).isNotNull();
-        Assertions.assertThat(solverInstance.getScoreWrapper().getScoreString()).isNotNull();
-        Assertions.assertThat(solverInstance.getScoreWrapper().getScoreString()).isNotEmpty();
+        AwaitilityUtils.untilIsNotNull(() -> solverClient.createSolver(containerId, CLOUD_BALANCE_SOLVER_ID, CLOUD_BALANCE_SOLVER_CONFIG));
+
+        AwaitilityUtils.untilIsNotNull(() -> {
+            try {
+                solverClient.solvePlanningProblem(containerId, CLOUD_BALANCE_SOLVER_ID, loadPlanningProblem(kieContainer, 5, 15));
+            } catch (NoSuchMethodException | ClassNotFoundException | IllegalAccessException | InstantiationException | InvocationTargetException e) {
+                throw new RuntimeException(e);
+            }
+
+            return solverClient.getSolver(containerId, CLOUD_BALANCE_SOLVER_ID); // wait for 15 seconds
```

Comment:
There is no waiting, removing comment.
```suggestion
            return solverClient.getSolver(containerId, CLOUD_BALANCE_SOLVER_ID);
```